### PR TITLE
sessionInfo GUI for badchannels (#120)

### PR DESCRIPTION
**Pull request by dlevenstein**
bz_getSessionInfo(basePath,’editGUI’,true) can now edit badchannels and region…. try it out, make some improvements, and let’s merge it into master….

:D

### CODE REVIEW

**Comment by brendonw1** [comment]
We might try it Monday actually... I think

**Comment by brendonw1** [comment]
There is a missing dependency: StructDlg.m.  I think we either somehow tell people to download it or we make a new bz_StructDlg.m that is identical and credits the original (with website).  What do you think?

Actually we also use v2struct, which is downloaded code.

What should we do with such things?

**Comment by dlevenstein** [comment]
Structdlg is in the external packages folder?

**Comment by brendonw1** [comment]
Ah it might be my fault then... I didn't addpath(genpath(new version of
repo))

**Comment by brendonw1** [comment]
So it's fine if I add extra info to the session info right?  The structure
I put together had much more info that I find useful (ie turns, Z location,
probe type etc).  That's fine right?
Also I wonder how this fits in with having sufficient metadata to fit with
neurodata without borders - or even just a larger more inclusive set of
metadata?  This seems insufficient now right?  Will we try to address that?

**Comment by dlevenstein** [comment]
Yes, I think at this point this is not sufficient metadata to be a full record of a recording. And we would like to address that... Now seems to be a good time to start making a metadata file that can be both convenient for everyday analysis and extensive enough for long-term storage. I.e. by combining the more extensive metadata you've set up into the sessionInfo file.

So far, we've tried to maintain backwards compatibility with the structure that comes out of LoadParameters from the xml, which we should still do. And can then add on additional metadata in the form of substructures

Here's what I would propose as a starting point:
1) Maintaining top level fields as the basic things you'll need during analysis (some of these are depreciated, and should be removed at some point).
2) The more extensive details in substructures. Maybe the way to start this is to port over the whole ExtracellEphys field from your SessionMetadata file to the SessionInfo? Many of the things that are currently in top level can be moved into substructures.
3) Update the bz_sessionInfoGUI to be able to edit/enter/view these fields? My hope is that this can grow to a fully functional (and not ugly) GUI for viewing and editing the sessionInfo file, similar to how NDManager can view the .xml. 


@DavidTingley what do you think?

TL;DR - yes, add to the sessionInfo to fit your needs, but make sure that it's well-documented what the fields/structure are and that you update the wiki (and don't yet break backwards compatibility), and preferably add the mechanism for view/editing to bz_sessionInfoGUI via bz_getSessionInfo(basePath,’editGUI’,true)

**Comment by brendonw1** [comment]
Just to clarify, I know that what was done was done in an effort to bring
together the datasets and be convenient. I'm just pushing us to get it
right before we start getting to far down the road.  Do we put all that
metadata somewhere?  If so where?  Or do we each come up with our own
solution for that?  I think it's best if it's included in buzcode and I'm
happy to help do that, but my hestiancy comes from the fact that i already
made a bit effort to do that and it wasn't received well so I'd want to
hear what others want to do before doing anything.  It's not that I'm
asking others to do it, I'm happy to help, but I think this is something we
need consensus on and then we should move on it soon

**Comment by brendonw1** [comment]
We wrote at the same time, but it sounds like we're thinking similarly.  I
agree.

We could have sessioninfo and metadata be separate if we want.  I think
it's better to have them together but not necessary if that messes things
up.

But yeah we should get consensus before building tons of stuff on top of
this and then realizing we're missing a lot

Happy to take input about how to do it though and I really don't care
anymore if we use my old structures/codes - as long as it works and it's
sufficient

**Comment by dlevenstein** [comment]
There are two needs
-Ease of access to the relevant info during analysis (without having to go 5 fields deep)
-Storage of all metadata needed to keep good records.

My view would be to these are ideally in the same file (baseName.sessionInfo.mat). With things needed for everyday analysis given a privileged position as top-level fields, and extensive metadata storage in substructures.
And a unified/flexible GUI to view/enter metadata as needed.

I think combining your structures into the sessionInfo file, and making them viewable/enterable with sessioninfoGUI, is a good way to start towards that. Making it perfect will take some time/work/iterations as we use and improve it, but it's time to get it into one place so we can all be working on the same thing.

**Comment by dlevenstein** [comment]
In other news though, if you tried this and it works for you for entering/editing bad channels/regions, want to approve the pull request? :D 

The addition of the ECellEphys substructure should be it's own pull request/branch so we can all try it out etc.

Then we're just waiting on @raswanson ;)

**Comment by brendonw1** [comment]
Merged.

Do you think we should go gui-style for the other ExtraCellEphys stuff?

Also, what are we waiting on @raswanson for?

**Comment by dlevenstein** [comment]
Thanks! 

Yeah, my preference would be one-stop GUI for all entering/viewing of the sessionInfo metadata.

Was waiting on Rachel to try out the new code and approve the changes :p  But now we're merged so she missed her boat, eh?
